src/app/codegen-data-model-provider/Instance.cpp

```diff
@@ -14,15 +14,22 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
+#include "lib/core/CHIPPersistentStorageDelegate.h"
 #include <app/codegen-data-model-provider/CodegenDataModelProvider.h>
 #include <app/codegen-data-model-provider/Instance.h>
 
 namespace chip {
 namespace app {
 
-DataModel::Provider * CodegenDataModelProviderInstance()
+DataModel::Provider * CodegenDataModelProviderInstance(PersistentStorageDelegate * delegate)
```

Comment:
Having this mutate the state of the data model provider on each get is _really_ weird.  Why are we doing that?

It seems like the fundamental issue is that we have this global getter at all, instead of just letting people who want a CodegenDataModelProvider instantiate a CodegenDataModelProvider.

Reply:
I could not come up with a saner "object is defined elsewhere" solution. 
Technically this could be part of the constructor, but only if applications are responsible to allocate such an object.

Would accept any proposals to make this different.

Reply:
> but only if applications are responsible to allocate such an object.

Right.  Why are we not doing that?